### ISSUE TITLE
Add Pie Chart Key

### ISSUE THREAD

**Issue by GeorgeBerdovskiy** [PureChart/purechart#23]
## Background
PureChart supports simple pie charts due to #21, but they aren't very helpful yet because there is no key displaying the categories and their respective percentages. Below is a screenshot of what this can look like (taken from [monday.com](https://monday.com/)).

![image](https://github.com/PureChart/purechart/assets/43429820/150748e4-4bdc-4040-8c2b-b9675ecfee92)

## Goal
When rendering a pie chart, also render a key to the side of it displaying every category, its color, and its percentage. Note that this will involve adding an additional field to every data point, like `name` or `category`.

Although the pie chart is rendered as an SVG, using HTML/CSS for the key will actually be much easier! For this reason, the pie chart partial (`purechart/app/views/_pie.html.erb`) may come in handy. Here's what I'd recommend trying -
- Instead of rendering the rendered result (`result.html_safe`), pass it to the partial as a local variable
- Inside the partial, stick the SVG result onto the page using a template tag `<%= ... %>` and use HTML + inline CSS to generate the key

Feel free to delete anything inside `_pie.html.erb` as it's not being used. Good luck!

**Comment by GeorgeBerdovskiy** [comment]
Note - the data should be formatted like this.
```rb
[
  {
    value: 25,
    color: "#eb4d4b"
  },
  {
    value: 100,
    color: "#be2edd"
  },
  ...
]
```

**Comment by yooian** [comment]
If I'm allowed to I'd like to do this one as well (while waiting on pull request)

**Comment by GeorgeBerdovskiy** [comment]
Go ahead @yooian, since you're the pie expert here 🫡

**Comment by GeorgeBerdovskiy** [comment]
**Update -** work in progress by @yooian, keeping this one reserved even though two weeks have expired.

**Comment by GeorgeBerdovskiy** [comment]
Closed by #42
